mlrun/model_monitoring/stream_processing_fs.py

```diff
@@ -135,249 +107,383 @@ def __init__(
-        )
-        feature_set.graph.add_step(
-            "storey.steps.SampleWindow",
-            name="sample",
-            after="Aggregates",
-            window_size=self.sample_window,
-            key=ENDPOINT_ID,
-        )
-        # kv
-        feature_set.graph.add_step(
-            "ProcessBeforeKV", name="ProcessBeforeKV", after="sample"
-        )
-        feature_set.graph.add_step(
-            "WriteToKV",
-            name="WriteToKV",
-            after="ProcessBeforeKV",
-            container=self.kv_container,
-            table=self.kv_path,
-        )
-        feature_set.graph.add_step(
-            "InferSchema",
-            name="InferSchema",
-            after="WriteToKV",
-            v3io_access_key=self.v3io_access_key,
-            v3io_framesd=self.v3io_framesd,
-            container=self.kv_container,
-            table=self.kv_path,
-        )
-        # tsdb
-        feature_set.graph.add_step(
-            "ProcessBeforeTSDB", name="ProcessBeforeTSDB", after="sample"
-        )
-        feature_set.graph.add_step(
-            "FilterAndUnpackKeys",
+    def apply_monitoring_serving_graph(self, fn):
+        """
+        Apply monitoring serving graph to a given serving function. The following serving graph includes about 20 steps
+        of different operations that are executed on the events from the model server. Each event has
+        metadata (function_uri, timestamp, class, etc.) but also inputs and predictions from the model server.
+        Throughout the serving graph, the results are written to 3 different databases:
+        1. KV (steps 7-9): Stores metadata and stats about the average latency and the amount of predictions over time
+           per endpoint. for example the amount of predictions of endpoint x in the last 5 min. This data is used by
+           the monitoring dashboards in grafana. Please note that the KV table, which can be found under
+           v3io:///users/pipelines/project-name/model-endpoints/endpoints/ also contains data on the model endpoint
+            from other processes, such as current_stats that is being calculated by the monitoring batch job
+            process.
+        2. TSDB (steps 12-18): Stores live data of different key metric dictionaries in tsdb target. Results can be
+           found under v3io:///users/pipelines/project-name/model-endpoints/events/. At the moment, this part supports
+           3 different key metric dictionaries: base_metrics (average latency and predictions over time),
+           endpoint_features (Prediction and feature names and values), and custom_metrics (user-defined metrics).
+           This data is also being used by the monitoring dashboards in grafana.
+        3. Parquet (steps 19-20): This Parquet file includes the required data for the model monitoring batch job
+           that run every hour by default. The parquet target can be found under
+           v3io:///projects/{project}/model-endpoints/.
+
```

Comment:
well done! I can see the amount of effort you put into it, be sure that  the docstring and the separation to steps will be very useful for any developer diving into this code :)